### ISSUE TITLE
Navigation-Guided Sparse Scene Representation for End-to-End Autonomous Driving

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#67]
### Problems of Perception Modules
- Most current E2EAD approaches build upon **complex perception frameworks**, often incorporating additional NN-based planning modules.
- These approaches typically inherit tasks such as object detection, mapping, occupancy prediction, resulting in **large and computationally intensive** neural networks.
- These models still maintain modular framework design, requiring **independent sub-tasks supervision**.
- As a result, they remain **annotation-intensive**, suffer from **scalability issues**, and are **inefficient for real-time deployment**.

### Questions

- Do E2EAD systems still require such extensive perception tasks?

- How can we efficiently identify and focus on the crucial parts of the scene without auxiliary perception supervision?

<img width="1431" alt="Image" src="https://github.com/user-attachments/assets/543aa15b-6b71-4c68-8f06-08454afa3d11" />

- SSR directly extracts only the essential perception elements in the guidance of navigation commands ((e.g., “turn left”, “turn right”, “go straight”), thereby minimizing redundancy. 
- This approach takes full advantage of the end-to-end framework, breaking the modular cascade architecture in a **Navigation-Guided Perception** manner. 
- Prior works attempt to reduce computation by skipping BEV feature construction, they still depend on **hundreds** of task-specific queries. However, our method drastically reduces computational overhead by using just **16** tokens guided by navigation commands.

<img width="1258" alt="Image" src="https://github.com/user-attachments/assets/1ff6a03a-541c-4e64-86c9-2ea4d06d0ebe" />

### Task Formulation 

At timestamp t, inputs are N-views images and a high-level navigation command cmd. The vision-based E2EAD model aims to predict the planning trajectory T, which consists of a set of points in BEV space. 

<img width="1279" alt="Image" src="https://github.com/user-attachments/assets/d31b42aa-6c67-4a5f-bc52-152a4a9d09c8" />

- BEV Encoder is BEVformer
- Scenes TokenLearner (STL) module is proposed to exteact scene queries from BEV feature. 

<img width="1325" alt="Image" src="https://github.com/user-attachments/assets/7a2d9766-bacd-4644-8dc1-20d6ced028ee" />

- Squeeze-and-Excitation (SE) layer is used to encode navigation command cmd into the dense BEV feature, producing the navigation-aware BEV feature. 
- The navigation-aware BEV feature is then passed into the BEV TokenLearner to adaptively focus on the most important information.

> **Squeeze-and-Excitation (SE)** is a tiny module that learns to emphasize important feature channels and suppress less useful ones. First, it “squeezes” each channel into a single value by taking the average over the entire feature map. Then, it runs those values through a small MLP to figure out which channels are most relevant, and “excites” those channels by giving them bigger weights. In the SSR framework, SE is used to combine navigation commands (like going straight or turning) with the BEV features. SSR applies a Squeeze-and-Excitation (SE) layer to fuse that command embedding with the BEV feature. Essentially, the command is used to generate channel-wise weights that highlight portions of the feature map relevant to the current maneuver. 

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/b7b33e32-4006-42d0-beda-8e6f25c592ba" />

> **TokenLearner** is a small module designed to select the most valuable “tokens” from a large pool of features. Essentially, it performs an attention-based filtering step: each region in the feature map is assigned a learned attention score, and only the regions with the highest scores are “kept” as tokens. This process significantly reduces computational load by focusing on a compact set of crucial features. In **SSR**, TokenLearner is applied to compress BEV (bird’s-eye-view) features. Guided by the driving command (e.g., turn left, go straight), it learns which parts of the BEV map are most relevant for the current navigation task, extracting only a small number of high-importance tokens. This approach allows SSR to plan trajectories efficiently without wasting effort on less critical information.

## Planning Phase

- In SSR’s planning phase, the model starts with 16 scene tokens that summarize the environment’s key information. 
- It also maintains a set of **waypoint queries** (learnable), each corresponding to a future position of the vehicle. 
- These queries perform **cross-attention** against the 16 scene tokens, extracting the most relevant details for trajectory generation. 
- Finally, each enriched query is passed through an MLP to predict the vehicle’s future coordinates, thus forming the overall planned trajectory.

## Temporal Enhancement by Future Feature Predictor 

**1. Predicting Future Scene Features**  
After generating the current trajectory, SSR uses it to predict the “next-frame BEV features” through a **Future Feature Predictor**. The rationale is simple:  
> If the current trajectory is accurate, then the future scene we observe should align with the predicted “future scene” based on that trajectory.

**2. Merging Trajectory Information (Dreaming Queries)**  
First, the 16 scene tokens from the current frame are combined with the predicted trajectory \(T\) via a Motion Aware Layer Normalization (MLN). The MLN injects trajectory information into the tokens, creating “Dreaming Queries.” This step ensures that the tokens now carry knowledge of “where we plan to go.”

**3. Using Dreaming Queries to Predict the Next-Frame BEV**  
These Dreaming Queries then pass through another module (with multi-layer self-attention) to produce “predicted next-frame scene tokens.” Finally, a TokenFuser reconstructs them back into a dense BEV feature map, denoted by \(\hat{B}_{t+1}\).

> **How TokenFuser Works in SSR** 1. **Weight Map Generation:**  
   TokenFuser starts by passing the current BEV feature \(B_t\) (shape \([H,W,C]\)) through a small MLP with a sigmoid. This produces a weight map of shape \([H,W,N_s]\), where \(N_s\) is the number of predicted scene tokens. Each pixel thus has a set of weights indicating how much it “relies on” each token. 2. **Token Fusion:**  
   Let the predicted next-frame scene tokens be \(\hat{S}_{t+1}\) (shape \([N_s,C]\)). The TokenFuser multiplies these tokens with the \([H,W,N_s]\) weight map (broadcast over the token and spatial dimensions). As a result, each spatial location in \([H,W]\) effectively “blends” the tokens together, weighted by those learned attention scores. 3. **Reconstructing the Next BEV:**  After the multiplication, the combined result is reshaped (or summed/reduced as needed) back to \([H,W,C]\), producing the predicted next-frame BEV feature \(\hat{B}_{t+1}\). This feature is then compared to the ground-truth BEV feature \(B_{t+1}\) via an L2 loss, enabling self-supervised learning of future scene dynamics.

**4. Self-Supervision (Comparing with the Real Next Frame)**  
\(\hat{B}_{t+1}\) is then directly compared to the actual future BEV feature \(B_{t+1}\) using an L2 loss. Any discrepancy triggers backpropagation, helping the model learn how the scene should evolve if it follows the predicted trajectory—no explicit detection or segmentation labels required.

Through this pipeline, SSR leverages temporal context to refine its scene representation and trajectory predictions, learning robust spatiotemporal features without reliance on supervised perception tasks.

## Experiments 

<img width="1022" alt="Image" src="https://github.com/user-attachments/assets/5dbb599b-09ef-47ae-810c-f7bb3a033343" />

<img width="1243" alt="Image" src="https://github.com/user-attachments/assets/2e3700fe-b4e4-43cb-8480-c456c9f1bbf1" />

<img width="1257" alt="Image" src="https://github.com/user-attachments/assets/2170223f-0a6d-4184-b29e-e2d4cc70a351" />

<img width="1242" alt="Image" src="https://github.com/user-attachments/assets/c5e07f3d-0442-4d61-8a0f-5a739f4f9b6b" />
